openapi-generator/src/main/kotlin/org/jellyfin/openapi/builder/openapi/OpenApiApiServicesBuilder.kt

```diff
@@ -63,6 +64,39 @@ class OpenApiApiServicesBuilder(
 		else -> null
 	}
 
+	private fun buildRequestBody(requestBody: RequestBody, serviceName: String, operationName: String): TypeName? {
+		// Filter out duplicate JSON values
+		val requestBodyTypes = requestBody.content.keys.filterNot {
+			MimeType.IGNORED_JSON_TYPES.contains(it)
+		}
+
+		// Check amount of types
+		if (requestBodyTypes.size > 1) throw OpenApiGeneratorError("Multiple request body types are unsupported.")
+		else if (requestBodyTypes.isEmpty()) return null
+
+		// Retrieve info
+		val requestBodyType = requestBodyTypes.first()
+		val content = requestBody.content[requestBodyType]
+		val required = requestBody.required ?: false
+
+		// Determine the correct type
+		return when (requestBodyType) {
+			// JSON body - use type builder
+			MimeType.APPLICATION_JSON -> content?.schema?.let { schema ->
+				openApiTypeBuilder.build(
+					ApiTypePath(serviceName, operationName, ApiTypePath.PARAMETER_BODY),
+					schema
+				).copy(nullable = !required)
+			}
+			// Image body
+			MimeType.IMAGE_ALL -> Types.BYTE_ARRAY
```

Comment:
This woudln't handle something like `image/png`, would it? Iirc Ktor has a MimeTypes parser/matcher that can also match the 'all' type properly. Not sure whether that could be necessary for us in the future (at least right now it apparently isn't).

Reply:
You're correct, the server spec always uses `image/*` right now so I used a strict check.

Reply:
Alright, just something to keep in mind for the future then.